include/seqan3/core/metafunction/basic.hpp

```diff
@@ -23,6 +23,59 @@ namespace seqan3
 /*!\addtogroup metafunction
  * \{
  */
+ // ----------------------------------------------------------------------------
+ // delete_const
+ // ----------------------------------------------------------------------------
+
+/*!\brief Return the input type with the topmost `const` removed [Type metafunction].
+ * \tparam t The type to operate on.
+ * \attention In contrast to std::remove_const, this function also removes `const` from pointers and references.
+ */
+template <typename t>
+struct delete_const
+{
+    //!\brief The return type is the input type with the topmost `const` stripped.
+    using type = t;
+};
+//
+//!\overload
```

Comment:
I don't think //!overload works for metafunctions... have you verified how the documentation looks? For simple metafunctions like this, if you can't make it appear nice, just hide the "overloads" with \cond and clearly document the behaviour in the main implementation